### DISCUSSION TITLE
Server Freeze on Cache Destroy

### DISCUSSION THREAD

**Discussion by Legion4405** [Vdauphin/HeartsAndMinds#1587] [Question and answer] [answered]
Hello there, I was hoping someone could help me with this.

Our server is running Hearts and Minds on Takistan, but when destroying a cache, the server seems to freeze right before it gets destroyed. Nothing stands out in the RPT from what I can see, as the server freezes the second it gets destroyed.

We've used AT Launchers, and demolition blocks to destroy the caches. Sometimes it's fine, sometimes it crashes. 
What could cause this, and what could fix this?

**Comment by Legion4405** [comment]
Happens on clientside as well, game freezes. Nothing interesting showing in RPT

**Comment by Legion4405** [comment]
Did some further debugging, and got a fresh mission file from the github. Set on Altis, and tested it on there. Destroyed a couple of Ammocaches without any problem. All on clientside.
Got another fresh install of the mission, changed .Altis to .takistan. Blew up a cache, and my game crashed.

It seems that somehow Takistan causes the game to crash when blowing up a cache.. At least, it does for me, and the server my  community hosts.

**Comment by Legion4405** [comment]
Found the issue Core > FNC > Cache > find_pos.sqf L42, [] call btc_cache_fnc_find_pos seems to run in a loop if it can not find a suitable building to place a new Cache inside of. Putting: sleep 1; before btc_cache_fnc_find_pos gets called seems to resolve the problem.

**Reply by Vdauphin** [reply, marked as answer]
Hello,

Thank you a lot for this valuable analyse
The `sleep` will not work becauuse this code is executed in unscheduled environnement
This will should fix the issue: https://github.com/Vdauphin/HeartsAndMinds/pull/1588

Cheers

**Comment by Legion4405** [comment]
Happy to help!
